### ISSUE TITLE
Group reports

### ISSUE THREAD

**Issue by mattdavenport** [kriskbx/gitlab-time-tracker#4]
It would be nice to have a CLI command for generating group reports. For instance, providing `[namespace]` instead of `[namespace]/<project>` as args could generate a large report covering each project in the group. Subgroups could be included with an optional `--include-subgroups` flag.

**Comment by kriskbx** [comment]
Thanks for your suggestion!

I would keep the report command as it is and rather implement a new type option: `--type=group` and `--type=project`, maybe `--type=user` in the future. So you could write this:

```
gtt report groupname --type=group
gtt report username --type=user
```

Subgroups is kinda tricky: GitLabs API doesn't allow you to query subgroups directly. Instead you have to get all groups you're a member of and filter them by their `parent_id` respectively. This is very annoying. Nevertheless, I will work on it and let you know when I push changes.

**Comment by kriskbx** [comment]
I just pushed a first draft of the feature to `dev`, please check it out and tell me what you think: https://github.com/kriskbx/gitlab-time-tracker/tree/dev#query-groups-and-subgroups

**Comment by mattdavenport** [comment]
Looks great! Agreed on the points above regarding the interface. Thanks for the speedy reply/implementation!

**Comment by kriskbx** [comment]
You're welcome. :)

I'm looking into opening the report command itself for multiple projects or groups like this:
```
gtt report "user/project1" "user/project2"
gtt report group1 group2 --type=group
```
So I won't release the group feature until this is implemented. But you can use the dev branch until then.

**Comment by kriskbx** [comment]
I just published 1.4.0 with support for (multiple) groups and subgroups as well as multiple projects.
